### ISSUE TITLE
Add a custom query parameter to all Keycloak endpoints

### ISSUE THREAD

**Issue by andreapigatto** [AxaFrance/react-oidc#1089]
Hi, 
I'm trying to choose programatically the theme to use in Keycloak according to a query parameter that I pass on the frontend url of my SPA for branding purposes.
The idea would be to add a query param like `&theme=xx` to any of the Keycloak urls (login, register, etc.). Then using a custom SPI, retrieve this parameter and use it to override the ThemeSelector class of Keycloak setting programatically the theme.

This is actually working.

My issue is that I'm able to pass the query param to the Keycloak url only for the first /auth endpoint when I redirect from my SPA home address to Keycloak.
To do this I added this prop to the OidcProvider:
`extras: { theme: paramFromUrl }` where paramFromUrl is extracted from the url of my SPA.

This as said works only for the /auth endpoint but not to for all the others (for ex. when I click register from the login page, I'm redirected to the register page of Keycloak and I lose the query param)

Do you know if there is a way to set the query params for any of the keycloak endpoints? Or to inject some logic similar to a middleware before redirection?
With keycloak-js seems possible to replace the url before redirect defining a custom adapter as you see here in the keycloakify project: https://github.com/keycloakify/keycloakify-starter/blob/cb5844c62381efed7b303886cbe460c055a62c21/src/App/oidc.tsx#L66-L83

Thanks

**Comment by guillaume-chervet** [comment]
Hi @andreapigatto , thank you for your issue.
Did you use the extras parameters from your configuration ? 
https://github.com/AxaFrance/react-oidc/blob/3e535c94dc57d59956498bfa0ca9536106039bda/packages/react/src/configurations.ts#L17

It is not used from the logout where you need to pass params as input to logout function. It can be an enhcement to add.

**Comment by andreapigatto** [comment]
@guillaume-chervet yes I'm adding the query param I want to add on extras. But this is used only on the call to the initial `/auth` endpoint. Any other endpoint like `/registration` are not adding what I define in the extra field as custom query param.

**Comment by guillaume-chervet** [comment]
Hi @andreapigatto , do you have a video of you navigation or something like that?

React oidc does not have any notion of /registration endpoint. Is it internal to keycloak ?

**Comment by andreapigatto** [comment]
https://drive.google.com/file/d/1POQodoa4Z_RVLwvaspGEjV0qVuEICxCd/view?usp=drive_link
Basically I would need some kind of persistence of this query param cause at every call to Keycloak there will be the check of this according to what I developed in the custom SPI. So summarising:
- login screen: have query param --> custom theme applied
- other screens: doesn't have query param --> default theme applied

I guess is something that should happen on frontend cause in the Keycloak server I have no access to localStorage or anything that I can help to keep persistent the theme variable after the first call.

**Comment by guillaume-chervet** [comment]
hi @andreapigatto ,

Thank you very much for the video. It help a lot.

The register button you clicked come from KeyCload User Interface. I think you can configure it from keycloak side (not from react OIDC).

**Comment by andreapigatto** [comment]
@guillaume-chervet thanks for the feedback. Yes I checked even keycloak-js and the control is only on the first /auth endpoint. After should be managed somehow all in keycloak side.

**Comment by guillaume-chervet** [comment]
Hi @andreapigatto , did you succed ?  May we close the issue?

**Comment by guillaume-chervet** [comment]
Thank you for your feedback @andreapigatto ! Have fun :)
